cleverhans_tutorials/mnist_tutorial_tfe.py

```diff
@@ -0,0 +1,253 @@
+        error += attacks_list
+        raise AttributeError(error)
+    # Mapping attack from string to class.
+    attack_class = ListOfAttacks[attack_string]
+    return attack_class
+
+
+def mnist_tutorial(train_start=0, train_end=60000, test_start=0,
+                   test_end=10000, nb_epochs=6, batch_size=128,
+                   learning_rate=0.001,
+                   clean_train=True,
+                   testing=False,
+                   backprop_through_attack=False,
+                   nb_filters=64, num_threads=None,
+                   attack_string=None):
+    """
+    MNIST cleverhans tutorial
+    :param train_start: index of first training set example.
+    :param train_end: index of last training set example.
+    :param test_start: index of first test set example.
+    :param test_end: index of last test set example.
+    :param nb_epochs: number of epochs to train model.
+    :param batch_size: size of training batches.
+    :param learning_rate: learning rate for training.
+    :param clean_train: perform normal training on clean examples only
+                        before performing adversarial training.
+    :param testing: if true, complete an AccuracyReport for unit tests
+                    to verify that performance is adequate.
+    :param backprop_through_attack: If True, backprop through adversarial
+                                    example construction process during
+                                    adversarial training.
+    :param nb_filters: number of filters in the CNN used for training.
+    :param num_threads: number of threads used for running the process.
+    :param attack_string: attack name for crafting adversarial attacks and
+                            adversarial training, in string format.
+    :return: an AccuracyReport object
+    """
+
+    # Object used to keep track of (and return) key accuracies
+    report = AccuracyReport()
+
+    # Set TF random seed to improve reproducibility
+    tf.set_random_seed(1234)
+
+    # Set logging level to see debug information
+    set_log_level(logging.DEBUG)
+
+    # Get MNIST test data
+    X_train, Y_train, X_test, Y_test = data_mnist(train_start=train_start,
+                                                  train_end=train_end,
+                                                  test_start=test_start,
+                                                  test_end=test_end)
+
+    # Use label smoothing
+    assert Y_train.shape[1] == 10
+    label_smooth = .1
+    Y_train = Y_train.clip(label_smooth / 9., 1. - label_smooth)
+
+    # Train an MNIST model
+    model_path = "models/mnist"
+    train_params = {
+        'nb_epochs': nb_epochs,
+        'batch_size': batch_size,
+        'learning_rate': learning_rate
+    }
+
+    # Initialize the attack object
+    attack_class = attack_selection(attack_string)
+    attack_params = {
+        'eps': 0.3, 'clip_min': 0.,
+        'clip_max': 1.}
```

Comment:
Why put this on an extra line when you still have space on the previous one?

Reply:
True, removed the extra line.